eth/hooks/engine_v2_hooks.go

```diff
@@ -23,12 +24,16 @@ func AttachConsensusV2Hooks(adaptor *XDPoS.XDPoS, bc *core.BlockChain, chainConf
 	// Subnet penalty is triggered at gap block, and stopped at previous gap block
 	adaptor.EngineV2.HookPenalty = func(chain consensus.ChainReader, number *big.Int, currentHash common.Hash, candidates []common.Address, config *params.XDPoSConfig) ([]common.Address, error) {
 		start := time.Now()
+		listBlockHash := []common.Hash{}
+		listBlockNumber := []uint64{}
```

Comment:
For the 'list' naming, I would say just put 's' at the end of name, that would be readable.